API/APIManager.m

```diff
@@ -79,4 +69,48 @@ - (void)postStatusWithText:(NSString *)text completion:(void (^)(Tweet *, NSErro
     }];
 }
 
+- (void)favorite:(Tweet *)tweet completion:(void (^)(Tweet *, NSError *))completion {
+    NSString *urlString = @"1.1/favorites/create.json";
+    NSDictionary *parameters = @{@"id": tweet.idStr};
+    [self POST:urlString parameters:parameters progress:nil success:^(NSURLSessionDataTask * _Nonnull task, NSDictionary *  _Nullable tweetDictionary) {
+        Tweet *tweet = [[Tweet alloc]initWithDictionary:tweetDictionary];
+        completion(tweet, nil);
+    } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
+        completion(nil, error);
+    }];
+}
+
+- (void)unfavorite:(Tweet *)tweet completion:(void (^)(Tweet *, NSError *))completion {
```

Comment:
Is unfavorite really the same as delete / destroy ?

Reply:
Also, this is basically the same as favorite. Leading question, is there a good way to combine these two methods?